cookbooks/aws-parallelcluster-install/files/default/ec2_udev_rules/attachVolume.py

```diff
@@ -148,5 +121,79 @@ def main():
             continue
 
 
+def detach_volume(volume_id, ec2):
+    response = ec2.detach_volume(VolumeId=volume_id)
+
+    # Poll for volume to attach
+    state = response.get("State")
+    x = 0
+    while state != "available":
+        if x == 60:
+            print("Volume %s failed to detach in 300 seconds." % volume_id)
+            exit(1)
+        if state in ["busy", "attached"]:
+            print("Volume %s in bad state %s" % (volume_id, state))
+            exit(1)
+        print("Volume %s in state %s ... waiting to be 'detach'" % (volume_id, state))
+        time.sleep(5)
+        x += 1
+        try:
+            state = ec2.describe_volumes(VolumeIds=[volume_id]).get("Volumes")[0].get("State")
+        except IndexError:
+            continue
+
+
+def handle_volume(volume_id, attach, detach):
+    # Get IMDSv2 token
+    token = get_imdsv2_token()
+
+    # Get instance ID
+    instance_id = requests.get("http://169.254.169.254/latest/meta-data/instance-id", headers=token).text
+
+    # Get region
+    region = requests.get("http://169.254.169.254/latest/meta-data/placement/availability-zone", headers=token).text
+    region = region[:-1]
+
+    # Parse configuration file to read proxy settings
+    config = configparser.RawConfigParser()
+    config.read("/etc/boto.cfg")
+    proxy_config = Config()
+    if config.has_option("Boto", "proxy") and config.has_option("Boto", "proxy_port"):
+        proxy = config.get("Boto", "proxy")
+        proxy_port = config.get("Boto", "proxy_port")
+        proxy_config = Config(proxies={"https": "{0}:{1}".format(proxy, proxy_port)})
+
+    # Connect to AWS using boto
+    ec2 = boto3.client("ec2", region_name=region, config=proxy_config)
+
+    if attach:
+        attach_volume(volume_id, instance_id, ec2)
+    elif detach:
+        detach_volume(volume_id, ec2)
+
+
+def main():
```

Comment:
Minor: we should consider to rename this script into `manageEbsVolume.py` because it covers both attachment and detachment now and also because it is focused on EBS, not other types of volumes (FSx Ontap volumes)